packages/components/src/components/modal/modal.jsx

```diff
@@ -22,17 +22,20 @@ class ModalElement extends React.PureComponent {
         }
         this.el.classList.add('dc-modal');
         this.state.modal_root.appendChild(this.el);
+        this.props.onMount();
```

Comment:
After discussing with @mahdipourziaei-binary and @oskar-binary, let's add a check rather than defining `defaultProps` for this component. This will make the code more readable, as well as be more consistent with the rest of our components. e.g.

```JS
if (typeof this.props.onMount === 'function) {
    this.props.onMount();
}
```

Same can be applied for `onUnmount` later in your changes.

Reply:
sure. Thank you